pkg/apisix/cache/memdb.go

```diff
@@ -113,6 +127,8 @@ func (c *dbCache) GetUpstream(id string) (*v1.Upstream, error) {
 	if err != nil {
 		return nil, err
 	}
+	c.upstreamLock.Lock()
+	defer c.upstreamLock.Unlock()
```

Comment:
@liushv0 I think there is no point in having locks as well as DeepCopy. I think this line need to be removed https://github.com/apache/apisix-ingress-controller/blob/58aab33bc6d215c6a3f826daedbd5a5d85ca6c23/pkg/types/apisix/v1/zz_generated.deepcopy.go#L336. This line may have been added when this file was autto generated.

Reply:
I guess remove  this line is effective，and less change，
but honesty，I am not exactly understand  the intention of this line，set *out = *in for what。
other side，this line  is generated by code-gen tools，maybe somebody will rerun  code-gen tools，and overwrite the  modification。
that is why I choose use  locks。